lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -721,6 +721,7 @@ def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
             return
         self.peers_data.delete(Scope.UNIT, "started")
         if self.opensearch.is_started():
+            self.unit.open_port("tcp", 9200)
```

Comment:
This would only apply in case a previous start had timed-out for ex [here](https://github.com/canonical/opensearch-operator/pull/249/files#diff-5c0d89f838ffd5a9ecf2288b3698320fda2d2b2b737361b6e10adb55d1929f43R773). A safer place would be at the end of `_post_start_init`, as it is guaranteed to be invoked after the current opensearch service stabilizes on the first try or a subsequent retry

Reply:
Thanks for the comment, Mehdi, I got this suggestion also from Pedro, and it's now moved to _post_start_init.